src/sentry/api/endpoints/organization_dashboards.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import absolute_import
+
+from django.db import IntegrityError, transaction
+from sentry.api.base import DocSection
+from sentry.api.bases.organization import OrganizationEndpoint
+from sentry.api.bases.dashboard import DashboardSerializer
+from sentry.api.serializers import serialize
+from sentry.models import Dashboard
+from rest_framework.response import Response
+
+
+class OrganizationDashboardsEndpoint(OrganizationEndpoint):
+    doc_section = DocSection.ORGANIZATIONS
+
+    def get(self, request, organization):
+        """
+        Retrieve an Organizations Dashboards
+        `````````````````````````````````````
+         Retrieve a list of dashboards that are associated with the given organization.
+         :pparam string organization_slug: the slug of the organization the
+                                          dashboards belongs to.
+        :qparam string query: the title of the dashboard being searched for.
+        :auth: required
+        """
+        dashboards = Dashboard.objects.filter(
+            organization_id=organization.id
+        )
+        query = request.GET.get('query')
+        if query:
+            dashboards = dashboards.objects.filter(
+                title=query,
+            )
+
+        context = serialize(list(dashboards), request.user)
+        return Response(context)
+
+    def post(self, request, organization):
+        serializer = DashboardSerializer(data=request.DATA, context={'organization': organization})
+
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=400)
+
+        result = serializer.object
+
+        try:
+            with transaction.atomic():
+                dashboard = Dashboard.objects.create(
+                    organization_id=organization.id,
+                    title=result['title'],
+                    owner=result['owner'],
+                    data=result.get('data'),
+                )
+        except IntegrityError as e:
+            return Response(e.message, status=208)
```

Comment:
2xx are usually success statuses, we should probably return 400?

Reply:
208 is a webdav extension as well. I think a 409 conflict is what you want.

Reply:
Hey @wedamija and @markstory I was following the example here: https://github.com/getsentry/sentry/blob/master/src/sentry/api/endpoints/organization_releases.py#L264

Reply:
Ok can change it :)